### PR TITLE
Haskell: bump packages, close bugs

### PR DESCRIPTION
**Pull request by solpeth**
One linked bug will not auto-close, and requires retesting: https://bugs.gentoo.org/863191

### CODE REVIEW

File: dev-haskell/semigroupoids/metadata.xml [hunk 1]
```diff
@@ -5,45 +5,4 @@
...
-		</flag>
-		<flag name="unordered-containers">
-			Enable the use of the `unordered-containers` package (and also its
-			dependency `hashable`). Disabling this is an unsupported configuration,
-			but it may be useful for accelerating builds in sandboxes for expert users.
-		</flag>
-
-	</use>
-	<upstream>
-		<remote-id type="github">ekmett/semigroupoids</remote-id>
```
Comment (thesamesam) [inline dev-haskell/semigroupoids/metadata.xml:47]: :(

Comment (solpeth) [reply dev-haskell/semigroupoids/metadata.xml:47]: I'll turn that frown upside down, one sec

**Comment by solpeth** [comment]
ping @thesamesam - just wondering if this is okay to be merged. It will require retesting of some of the tinderbox bugs to see if this fixes them.

Thanks again!

**Comment by thesamesam** [comment]
Thank you!

**Pull request closed without merging**